libarchive/archive_read_open_filename.c

```diff
@@ -372,6 +370,10 @@ file_open(struct archive *a, void *client_data)
 		mine->use_lseek = 1;
 
 	return (ARCHIVE_OK);
+done_error:
+	if (0 < fd)
```

Comment:
I do not disagree but I cannot blame neither. @yaneurabeya just kept the style @kientzle used in the file. open() returns -1 on failure. If we want to change this than it should be done thoroughly in all similar cases in the whole file (and there are such cases).